src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -98,7 +98,7 @@ public async Task<ConnectNewHeadersResult> ConsensusTipChangedAsync()
             {
                 if (this.cachedHeaders.Count != 0 && this.CanConsumeCache())
                 {
-                    result = await this.PresentHeadersLockedAsync(this.cachedHeaders, false).ConfigureAwait(false);
```

Comment:
If we make that method none async we probably can remove the `asyncLock` it will make this PR bigger then needed.

Reply:
We can do in another PR.